src/Microsoft.Azure.ServiceBus/Core/IMessageReceiver.cs

```diff
@@ -133,6 +133,17 @@ public interface IMessageReceiver : IReceiverClient
         /// </remarks>
         Task RenewLockAsync(Message message);
 
+        /// <summary>
+        /// Renews the locks on the messages. The locks will be renewed based on the setting specified on the queue.
+        /// </summary>
+        /// <remarks>
+        /// When messages are received in <see cref="ServiceBus.ReceiveMode.PeekLock"/> mode, the messages is locked on the server for this
+        /// receiver instance for a duration as specified during the Queue/Subscription creation (LockDuration).
+        /// If processing of the messages requires longer than this duration, the lock needs to be renewed.
+        /// For each renewal, it resets the time the message is locked by the LockDuration set on the Entity.
+        /// </remarks>
+        Task RenewLocksAsync(IEnumerable<Message> messages);
```

Comment:
Note: this will be a breaking change that will require the feature to be released with the next major if feature is accepted.

Reply:
Alternatively, it could be added to `IMessageReceeiver` implementations and `IMessageReceiver` change would be added as a separate PR for the next major, similar to #486.